sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -27,10 +31,46 @@ def __init__(self):
         super().__init__()
 
         self._forecaster = None
+        pred_supported = self._check_supports_pred_int()
+        self._support_pred_int_in_sample = pred_supported["int_in_sample"]
+        self._support_pred_int = pred_supported["int"]
 
-    def _instantiate_model(self):
+        self.set_tags(
+            **{"capability:pred_int:insample": self._support_pred_int_in_sample}
+        )
+        self.set_tags(**{"capability:pred_int": self._support_pred_int})
+
+    def _get_statsforecast_class(self):
         raise NotImplementedError("abstract method")
 
+    def _get_statsforecast_params(self):
+        raise NotImplementedError("abstract method")
+
+    def _get_init_statsforecast_params(self):
+        statsforecast_class = self._get_statsforecast_class()
+        return list(signature(statsforecast_class.__init__).parameters.keys())
+
+    def _validate_init_params(self, **sktime_params):
+        sktime_params = sktime_params.copy()
+        statsforecast_params = self._get_init_statsforecast_params()
+
+        for sktime_param in sktime_params.keys():
+            if sktime_param not in statsforecast_params:
+                sktime_params.pop(sktime_param)
+                warn(
+                    f"Keyword argument '{sktime_param}' will be omitted as it is"
+                    f" not found in the __init__ method "
+                    f"from {self._get_statsforecast_class()}. "
+                    f"Check your statsforecast version"
+                    f"to find out the right API parameters."
+                )
+        return sktime_params
+
+    def _instantiate_model(self):
+        cls = self._get_statsforecast_class()
+        params = self._validate_init_params(**self._get_statsforecast_params())
```

Comment:
if we always call `_validate_init_params` with `**self._get_statsforecast_params()`, should it have an arg at all?

E.g., you could remove the arg, and rename the `_validate_init_params` to `_get_validated_statsforecast_params`

Reply:
done